docs/700_CONFIG.md

```diff
@@ -0,0 +1,18 @@
+# Config
+
+The config should be an abstraction for getting specific config values from
+different versions of an off-chain config.
+
+```
+type IConfig interface {
+    // Has simply indicates that the config has a specific value or not
+    Has(string) bool
+    // Get provides the means to get any value of any type from a configuration
+    // and an error can indicate type issues
+    Get(string, interface{}) error
+}
+```
```

Comment:
Why do we need an abstraction here instead of a fixed type?

Reply:
I'll think more about this one. I wasn't completely sold on it myself. The problem is that configurations change a lot over time and ideally we need to version the configs just as much as we need to version the observations. Each node could get a config that it isn't expecting. If we go with types like `ConfigV1` and `ConfigV2`, functions that need a config would need to accept `interface{}` instead which doesn't provide type safety.

This design provides necessary functions while any specific config manages its own parsing and validation. It's more akin to getting query parameters in an API handler.

Reply:
We need to think about config management and how that will evolve in future. A simple way to start would be that every new config should have a default value so we don't need different types